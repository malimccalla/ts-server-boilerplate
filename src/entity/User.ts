import * as bcrypt from 'bcryptjs';
import {
  BaseEntity,
  BeforeInsert,
  Column,
  Entity,
  PrimaryGeneratedColumn,
} from 'typeorm';

@Entity('users')
export class User extends BaseEntity {
  @PrimaryGeneratedColumn('uuid') id: string;

  @Column('varchar', { length: 255 })
  email: string;

  @Column('text') password: string;

  @Column('boolean', { default: false })
  confirmed: boolean;

  @Column('boolean', { default: false })
  locked: boolean;

  @BeforeInsert()
  async hashPassword() {
    const password = await bcrypt.hash(this.password, 10);
    this.password = password;
  }
}
