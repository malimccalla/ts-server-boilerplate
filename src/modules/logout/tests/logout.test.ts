import { Connection } from 'typeorm';

import { User } from '../../../entity/User';
import { createTestConn } from '../../../test/createTestConn';
import { TestClient } from '../../../test/TestClient';

const host = process.env.TEST_HOST as string;

describe('logout', () => {
  const email = 'cookie2@email.com';
  const password = 'password123';
  let conn: Connection;

  beforeEach(async () => {
    conn = await createTestConn();

    await User.create({
      email,
      password,
      confirmed: true,
    }).save();
  });

  afterEach(async () => {
    await conn.close();
  });

  test('logout a user', async () => {
    const client = new TestClient(host);
    await client.login(email, password);
    const { data } = await client.me();

    expect(data.me).toHaveProperty('email');
    expect(data.me).toHaveProperty('id');

    const res = await client.logout();
    const res2 = await client.me();

    expect(res.data).toBe(true);
    expect(res2.data.me).not.toHaveProperty('email');
    expect(res2.data.me).not.toHaveProperty('id');
    expect(res2.data.me).toBeNull();
  });
});
