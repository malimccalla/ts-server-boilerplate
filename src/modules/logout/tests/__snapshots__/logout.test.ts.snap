// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`logout logout a user 1`] = `
Object {
  "logout": true,
}
`;
